### ISSUE TITLE
filterProperties as Array doen't work

### ISSUE THREAD

**Issue by olasfar** [djett41/ionic-filter-bar#5]
$ionicFilterBar.show({
  items: items,
  update: function (filteredItems) {
    $scope.events = filteredItems;
  },
  filterProperties: ['toto', 'tata']
});

Doesn't return the right items.

With only one property it returns 3 items with 2 properties (['toto', 'tata']) it returns 0 items for the same keyword..

**Comment by djett41** [comment]
Thanks @olasfar .  If you can provide the following info, I can quickly reproduce the issue and provide a fix for you
1.  An example of the events array, along with an example of each event object in the array that your are filtering against. 
2.  A full example of how you are calling show.  For example, right now you are passing in a variable named items to the items property, but on update setting $scope.events.  Is that a typo or do you also have an items array?  If so what does the items array look like?
3.  The search text you are entering in the search box.

Thanks!

**Comment by djett41** [comment]
@olasfar I'm seeing expected behavior as described in the docs.  See the Angular docs on filter expression object's for more details ( https://docs.angularjs.org/api/ng/filter/filter )

For example, in the demo, if you were to add a foo property with a value of 'item' as below..

```
{
  text: 'This is item number ' + x + ' which is an ' + (x % 2 === 0 ? 'EVEN' : 'ODD') + ' number.', 
  foo: 'item'
}
```

Then in your call to show add foo as a second filter expression property as below

```
filterBarInstance = $ionicFilterBar.show({
    items: $scope.items,
    update: function (filteredItems) {
      $scope.items = filteredItems;
    },
    filterProperties: ['text', 'foo']
  });
```

And search for the text  'item', items in which have BOTH text AND foo properties that include the string 'item' will be filtered, which will be all of the items since both text and foo include the string 'item'.

If you were to then search for the text 'EVEN' you will notice that there are no results, that's because even though the text property will include the string EVEN for even items, there will never be an item in which text AND foo both have the text 'EVEN'

So to summarize, if you include filterProperties, then ALL of the properties will need to contain the search string to show the result.  

If you would like to have your item filtered if for example the text property OR foo property contained 'EVEN', then you would just remove the filterProperties option from your options object in your call to show(..).

Please let me know if this resolves your issue.

Thanks!!

**Comment by olasfar** [comment]
Thank you @djett41 for you help, it was just AND / OR issue :+1: Removing filterProperties resolves my issue.

**Comment by djett41** [comment]
Awesome @olasfar glad to hear!  :+1: :+1:
